R/cohenFleissKappa.R

```diff
@@ -0,0 +1,202 @@
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+cohenFleissKappa <- function(jaspResults, dataset, options) {
+  
+  ready <- length(options[["variables"]]) > 1
+  
+  dataset <- .cohensFleissKappaReadData(dataset, options)
+    
+  if(options[["cohensKappa"]])
+    jaspResults[["cohensKappa"]] <- .cohensKappa(dataset, options, ready)
+  
+  if(options[["fleissKappa"]])
+    jaspResults[["fleissKappa"]] <- .fleissKappa(dataset, options, ready)
+  
+  return()
+}
+
+# Read in the dataset (copied from .reliabilityReadData)
+.cohensFleissKappaReadData <- function(dataset, options) {
+  variables <- unlist(options[["variables"]])
+  if (is.null(dataset)) {
+    dataset <- .readDataSetToEnd(columns.as.factor = variables)
+  }
+  return(dataset)
+}
+
+.cohensKappa <- function(dataset, options, ready){
+  
+  weighted <- options[["cohensWeightedOrNot"]] == "cohensWeighted"
+  
+  if(weighted){
+    weightedString <- "Weighted"
+  }else{
+    weightedString <- "Unweighted"
+  }
+  
+  # Create the JASP Table
+  jaspTable <- createJaspTable(title = gettextf("Cohen's %s Kappa", casefold(weightedString, upper = F)))
+  jaspTable$addColumnInfo(name = "ratings", title = gettext("Ratings"), type = "string")
+  jaspTable$addColumnInfo(name = "cKappa", title = gettextf("%s Kappa", weightedString), type = "number")
+  jaspTable$position <- 1
+  
+  #dependencies
+  jaspTable$dependOn(
+    options = c(
+      "variables",
+      "cohensKappa",
+      "cohensWeightedOrNot",
+      "kappaIntervalOn",
+      "kappaConfidenceIntervalValue"
+    )
+  )
+  
+  
+  formattedCIPercent <- format(
+    100 * options[["kappaConfidenceIntervalValue"]],
+    digits = 3,
+    drop0trailing = TRUE
+  )
+  
+  
+  if(ready){
+    #Create all pairs
+    variables <- colnames(dataset)
+    allPairs <- .getPairs(variables)
+    allDataframes <- .getDataframes(dataset, allPairs)
+    allPairStrings <- paste(allPairs$x, allPairs$y, sep = " - ")
+    
+    #calculate Cohen's Kappas
+    allKappaData <- lapply(allDataframes, psych::cohen.kappa, alpha = 1 - options[["kappaConfidenceIntervalValue"]])
+    
+    #Extract Kappas and CIs
+    allKappas <- c()
+    allLowerBounds <- c()
+    allUpperBounds <- c()
+    
+    for(i in allKappaData){
+      kappaData <- i$confid
+      if(weighted){
+        k <- 2
+      }else{
+        k <-1
+      }
+      allKappas <- c(allKappas, kappaData[k,2])
+      allLowerBounds <- c(allLowerBounds, kappaData[k,1])
+      allUpperBounds <- c(allUpperBounds, kappaData[k,3])
+    }
+    
+    tableData <- list("ratings" = allPairStrings,
+                      "cKappa" = allKappas)
+    
+    if (options[["kappaIntervalOn"]]) {
+      jaspTable$addColumnInfo(name = "CIL", title = gettext("Lower"), type = "number", overtitle = gettextf("%s%% CI for Kappa", formattedCIPercent))
+      jaspTable$addColumnInfo(name = "CIU", title = gettext("Upper"), type = "number", overtitle = gettextf("%s%% CI for Kappa", formattedCIPercent))
```

Comment:
```suggestion
      jaspTable$addColumnInfo(name = "CIL", title = gettext("Lower"), type = "number", overtitle = gettextf("%s%% CI", formattedCIPercent))
      jaspTable$addColumnInfo(name = "CIU", title = gettext("Upper"), type = "number", overtitle = gettextf("%s%% CI", formattedCIPercent))
```

Reply:
Not sure what the convention is here? One sample t-test e.g., mentions what the CI refers to?

![image](https://user-images.githubusercontent.com/61797391/144716009-4dc0f4d4-843e-4793-9278-3705264d1e64.png)

Reply:
I asked EJ about this. If there is only one value the CI could refer to then it's fine to simply call it "CI".